src/main/java/com/icthh/xm/ms/entity/service/AttachmentService.java

```diff
@@ -44,13 +55,31 @@
      */
     @LogicExtensionPoint("Save")
     public Attachment save(Attachment attachment) {
-
+        Objects.nonNull(attachment);
+        Objects.nonNull(attachment.getXmEntity());
+        Objects.nonNull(attachment.getXmEntity().getId());
         startUpdateDateGenerationStrategy.preProcessStartDate(attachment,
                                                               attachment.getId(),
                                                               attachmentRepository,
                                                               Attachment::setStartDate,
                                                               Attachment::getStartDate);
-        attachment.setXmEntity(xmEntityRepository.getOne(attachment.getXmEntity().getId()));
+
+        XmEntity entity = xmEntityRepository.findById(attachment.getXmEntity().getId()).orElseThrow(
+            () -> new IllegalArgumentException("No entity found by " + attachment.getXmEntity().getId())
```

Comment:
"No entity found by id: "

Reply:
and better to throw  EntityNotFoundException then IllegalArgumentException
in this case 404 better than 500

Reply:
and better to throw  EntityNotFoundException then IllegalArgumentException
in this case 404 better than 500

Reply:
fixed